### ISSUE TITLE
Overlap of 2.5.1 Pointer Gestures and 2.5.7 Dragging Movements

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#3344]
If we have something like a path-based slider (where, as soon as you move your pointer out of the track, the slider "knob" doesn't follow the mouse anymore), 2.5.1 applies...but does 2.5.7? Or is the thinking that because it's path-based, it doesn't count as a pure "dragging movement", hence 2.5.7 does not apply?

and If so, should the definition for "dragging movement" be tweaked to make it clear that path-based movements (that still involve "dragging") are *excluded* from that definition?

https://w3c.github.io/wcag/guidelines/22/#dfn-dragging-movements

or should the normative wording for 2.5.7 be tweaked/clarified (probably too late now)

"All [functionality](https://w3c.github.io/wcag/guidelines/22/#dfn-functionality) that uses a [dragging movement](https://w3c.github.io/wcag/guidelines/22/#dfn-dragging-movements)" > "All [functionality](https://w3c.github.io/wcag/guidelines/22/#dfn-functionality) that uses a [dragging movement](https://w3c.github.io/wcag/guidelines/22/#dfn-dragging-movements) *that is not **path-based***"

**Comment by bruce-usab** [comment]
Bruce will try and find related conversation.  Dragging moving definition is very specific with mouse up/down.  So if following stops without mouse up, might not meet definition.

@patrickhlauke  we discussed on backlog call 8/18 -- can you provide another example?

**Comment by patrickhlauke** [comment]
having a hard time parsing that first sentence of yours, but to be clear: in general, any time you grab something and then have to follow a specific path and then release (sliders being the most obvious ones, when/if they are truly path-based and bail out of the interaction when you leave the slider track). it then is both a path-based interaction, and a dragging interaction, so assume that it then falls under *both* SCs. and if so, should be acknowledged in the respective understanding documents (happy to do a PR, but not until I hear if this take is "correct" based on other folks' understanding of these here)

**Comment by bruce-usab** [comment]
Also from call, adding example to Understanding with using finger as pointer would be helpful to make it clearer that SC is not only for mouse.

**Comment by bruce-usab** [comment]
@patrickhlauke in the conversation today, we focused on [the glossary definition](https://www.w3.org/TR/WCAG22/#dfn-dragging-movements), emphasis added:

> An operation where the pointer engages with an element on the down-event and the element ... follows the pointer **until an up-event** 

Agreed, sliders in some kind of track is common example.  The usual pattern, that bails out of the interaction when you leave the slider track, is not covered under 2.5.7.  Because there is no mouse up, there is no dragging movement.

> All functionality **that uses a dragging movement** for operation can be achieved by a single pointer without dragging.

As defined, within the context of 2.5.7, by cancelling the operation before the mouse up, that's _not_ using a dragging movement.  The 2.2 term, dragging movement, does not describe _all_ dragging interactions.  This distinction is part of how we were led to define "dragging movement" and not merely "dragging".

**Comment by patrickhlauke** [comment]
> Because there is no mouse-up, there is no dragging movement. From [the SC](https://www.w3.org/TR/WCAG22/#dragging-movements), again emphasis added:

there is an up movement if you pointer down, move along the track (to avoid it bailing), and then pointer up. that *is* a dragging movement.

**Comment by patrickhlauke** [comment]
i.e. this case https://codepen.io/patrickhlauke/pen/yMpbXa of a slider that:

* relies purely on dragging the slider knob
* doesn't provide a non-dragging way of setting a value
* is path-based (as it bails out when you stray out of the track while dragging)

fails both 2.5.1 *and* 2.5.7. which is fine, but needs to be explained/mentioned in understanding

**Comment by detlevhfischer** [comment]
@patrickhlauke It also bails out if you drag a bit too fast - I suggest to adapt the script so it won't bail out if there is no up event and the pointer remains within the track.

**Comment by patrickhlauke** [comment]
@detlevhfischer ah, noted, but that's beside the main point here...

**Comment by detlevhfischer** [comment]
I had assumed a core aspect of dragging movement was that it is not path-dependent, but the [definion of dragging movement](https://www.w3.org/TR/WCAG22/#dfn-dragging-movements) does not include that bit of disambiguation. It is only the [Dragging Movements Understanding doc](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html) that does this:
>  "In contrast, for dragging movements, only the start and end point of the movement matters, not the actual path"

**Comment by patrickhlauke** [comment]
Personally, I think it's better to accept that there might be a small overlap (where a path-based, drag-only widget fails both 2.5.1 and 2.5.7), and explaining *that* in the understanding, rather than trying to now redefine unintuitively that "dragging movement doesn't cover dragging movements that are also path-based", which would just open up a whole other can of worms making the SC inscrutable

**Comment by detlevhfischer** [comment]
Ah, gone back to the start of tie issue now, I see. My understanding was always that gestures and movements will be either / or (and the difficulty that content may behave differently on different UA / input methods has always been a problem here, especially concerning directionality, where it may seem required in a touch context, and not required in a mouse pointer context). Having said that, I think your codepen example would fail Dragging Movements and not Pointer Gestures since the thumb will just follow the pointer in any (both) directions within the given track - it is just a limitation of the field in which it can follow by constraining movement to a line. So there is no evaluation of duration, speed, or length of the engagement as you see it with typical pointer gestures (take those used for deleting mails or bringing up additional menu options like mark or file or "more")?

**Comment by patrickhlauke** [comment]
>  I think your codepen example would fail Dragging Movements and not Pointer Gestures since the thumb will just follow the pointer in any (both) directions within the given track - it is just a limitation of the field in which it can follow by constraining movement to a line

didn't we have lengthy conversations that this also falls under "path-based" ?

**Comment by detlevhfischer** [comment]
> didn't we have lengthy conversations that this also falls under "path-based" ?

Yes we had (have a hard time remembering how we got there though). The mere occurence of "dragging" in the above quote froom Understanding 2.5.1 is not helpful if we want more clarity here. The inclusion of sliders in 2.5.1 might have been driven by the wish to subume slider behavior under some SC available at that time. I had taken that position (that sliders fall under 2.5.1), but later, there were discussions around all these horizontal image sliders where the arument was made that they are technically implemented as dragging, then snapping to some position - and that directionality was not required if you ignore the fact that on a touch device intial vertical movement might pan the page rather than engage the slider. I have come round accepting that such sliders, at least, are better covered by Dragging Movements. Your code pen makes it hard to disambiguate due to the directional constraint (are there any such mplementations in the wild where the thumb gets dropped when you veer outside the track?) - and I see no problem failing those (rare?) cases in both SCs...

**Comment by patrickhlauke** [comment]
> are there any such implementations in the wild where the thumb gets dropped when you veer outside the track?

I've seen them very rarely, mainly from naive developers.

if the outcome of this discussion here is to *remove* those very specific sliders from 2.5.1 (which should just be a case of removing the example in 2.5.1's understanding), i could also live with that...

**Comment by bruce-usab** [comment]
@patrickhlauke and @detlevhfischer please provide feedback on [this version of Understanding 2.5.1](https://www.w3.org/WAI/WCAG21/Understanding/pointer-gestures.html) halfway down the page (sorry, no anchor) from
> There are times when a component requires a path-based gesture for touch screen devices but not with a mouse. Taking an example of a generic slider:

I think the three paragraphs pretty well describe when  slider uses dragging (or not).

**Comment by detlevhfischer** [comment]
@bruce-usab Due to the confusion of UA differences arising for the definition of „path-based gesture“ I find it preferable to change to a more narrowly constrained definition of what a path-based gesture is. 

The characteristics of the thumb slider and also image sliders that may not be triggered when moving a touch pointer vertically are such that in principle (depending on the position), you could move them in **either** horizontal direction, left **or** right. So that means they would not qualify as path-based if for path based, one would say it has to be one particular direction. 

So instead of saying, as now,  the path-based gesture has to go through an intermediate point (some point), one could define path-based as requiring a specific direction, and in addition optionally other characteristics like particular speed or length of path. If we go that route, control sliders with thumbs as well as image sliders would clearly be dragging-movement- rather than path-based-gesture-activated - regardless of UA, whether mouse or touch.

**Comment by patrickhlauke** [comment]
Tentatively revisiting this to try and clear some older issues, I think where I've landed at generally ... taking a big step back, both 2.5.1 and 2.5.7 want to make sure that interactions/interfaces can be operated with a single pointer, and don't require any particular dexterity (ability to move along a certain specific path, or speed/timing - only mentioned briefly once in 2.5.1's understanding, but should have been stressed further).

As awkward as the (quite artificial, in many ways) split is between 2.5.1 and 2.5.7, and where it overlaps, I think *together* the two SCs do cover the problem space reasonably well...but yes, to get there, they result in weird overlaps.

In the context of a WCAG 2.2 assessments, they give you the tools to flag problematic content quite well. But if doing just a 2.1 assessment, 2.5.1 required a bit of extra bending (that whole "passing through a mid-point"), retrofitting, retconning, finding slightly awkward justifications for including things under 2.5.1 that probably shouldn't have been. But it's probably too late now to try and fundamentally change this.

I think one way forward here is to quite clearly acknowledge the fact that 2.5.1 and 2.5.7 do have potential for overlap (and can lead to situations where something fails BOTH criteria), in the service of trying to cover all problematic scenarios. now that we do have 2.5.7, I might have another pass over 2.5.1 to see if some of the justification for including sliders under 2.5.1 could be removed, since the sliders would already fail 2.5.7 anyway in WCAG 2.2. this would remove some of the redundancy and confusion. and perhaps stressing the speed/timing aspect in 2.5.1 a bit more (there's currently one single use of the word "speed" in 2.5.1's understanding, mentioned briefly in passing)

**Comment by patrickhlauke** [comment]
I'm actually struggling now to remember what the point of including this was in 2.5.1 altogether ... that's the one that often causes confusion

> If going through an intermediate point (usually near the start of the gesture) affects its meaning, then it is a path-based gesture. The user engages a pointer (starting point), carries out a movement that goes through at least one intermediate-point before disengaging the pointer (end point). The intermediate point defines the gesture as requiring a specific path, even if the complete path is not defined.
> ![Hand showing a starting touch, 1. Moving through a second point, 2. Going to one of several points,3.](https://github.com/user-attachments/assets/d5395986-0880-48b5-914e-3a95e96b5317)
> Figure 2. A path-based gesture involves starting a pointer movement that goes through at least one intermediate point before the end-point. The end-point may be a continuation, or allow for various movements.

Which scenarios was that meant to cover, specifically? do you remember @alastc ? Was it primarily to hint at the fact that, say, a swipe's end point is usually not important, but that what counts is mainly the overall direction (and speed)?

Wondering if ripping this out, or replacing it with something that's clearer, would help.

**Comment by patrickhlauke** [comment]
@detlevhfischer I have made a start with ripping this up/apart a bit ... would love your initial reaction on it https://github.com/w3c/wcag/pull/4843

It's a draft PR for now, as I want to see if I can also cover/address other aspects such as https://github.com/w3c/wcag/pull/3148 https://github.com/w3c/wcag/issues/2116 and https://github.com/w3c/wcag/issues/2684
